app/models/scrape.rb

```diff
@@ -1,41 +1,57 @@
 class Scrape
   require 'open-uri'
-  def initialize(sub)
-    if sub == ""
-      @sub = "https://www.reddit.com/"
-    else
-      @sub = "https://www.reddit.com/r/#{sub}"
-    end
-    @textpage = Nokogiri::HTML(open(@sub))
-    @imagepage = Nokogiri::HTML(open('https://www.flickr.com/explore/interesting/7days/'))
+  
+  def initialize(subreddit_name)
+    @subreddit_name = subreddit_name
   end
 
-  def link
-    links = @textpage.css('div.link')
-    @link ||= links[rand(25)]
-    @link.css('a.title')
+  def post
+    Poster.new(text: subreddit_text, image: flickr_image)
+  end
+  
+  def reddit_textpage
+    @reddit_textpage ||= Nokogiri::HTML(open(reddit_uri))
+  end
+  
+  def flickr_imagepage
+    @flickr_imagepage ||= Nokogiri::HTML(open('https://www.flickr.com/explore/interesting/7days/'))
+  end
+
+  private
+  
+  def subreddit_link
+    @subreddit_link ||= subreddit_links.sample.css('a.title')
   end
 
-  def text
-    link.text
+  def subreddit_links
+    reddit_textpage.css('div.link')
+  end
+  
+  def subreddit_text
+    subreddit_link.text
   end
 
-  def url
-    @link.css('a.comments').map { |link| link['href'] }.first
+  # Unused method?
+  def subreddit_link_url
+    subreddit_link.css('a.comments').map { |link| link['href'] }.first
   end
```

Comment:
This parses the url for the comments page of the reddit post where the text came from. That's a mouthful.